### PR TITLE
MNT: Add the newport metadata

### LINKED ISSUE #781: Newport Motor Precision (and other metadata) Seen As Zero
## Expected Behavior

Newport motors should show a proper precision value so precision-aware applications can display their values properly.

## Current Behavior

This is always zero in our home-grown IOC when accessed the normal way ophyd uses to get these. See:
```
In [41]: xpp_lens_h.describe()
Out[41]:
OrderedDict([('xpp_lens_h',
              {'source': 'PV:XPP:LAS:MMN:06.RBV',
               'dtype': 'number',
               'shape': [],
               'units': '',
               'lower_ctrl_limit': 0.0,
               'upper_ctrl_limit': 0.0,
               'precision': 0}),
In [44]: xpp_lens_h.user_readback.metadata
Out[44]:
{'connected': True,
 'read_access': True,
 'write_access': False,
 'timestamp': 1615229826.060044,
 'status': <AlarmStatus.NO_ALARM: 0>,
 'severity': <AlarmSeverity.NO_ALARM: 0>,
 'precision': 0,
 'lower_ctrl_limit': 0.0,
 'upper_ctrl_limit': 0.0,
 'units': '',
 'enum_strs': None}
```
I checked a bunch of these, but here is the one from above:
```
$ caget -d 34 XPP:LAS:MMN:06.RBV
XPP:LAS:MMN:06.RBV
    Native data type: DBF_DOUBLE
    Request type:     DBR_CTRL_DOUBLE
    Element count:    1
    Value:            -1.775
    Status:           NO_ALARM
    Severity:         NO_ALARM
    Units:
    Precision:        0
    Lo disp limit:    0
    Hi disp limit:    0
    Lo alarm limit:   nan
    Lo warn limit:    nan
    Hi warn limit:    nan
    Hi alarm limit:   nan
    Lo ctrl limit:    0
    Hi ctrl limit:    0
```
```
$ caget XPP:LAS:MMN:06.PREC
XPP:LAS:MMN:06.PREC            3
```

## Possible Solution

Use the individual field PVs to fill in the metadata for Newport motors

## Context

- Makes the bluesky live table show single-digit precision
- Probably breaks typhos too

## Your Environment

latest

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Overwrite the incorrect metadata from the user_readback signal

## Motivation and Context

closes #781 

## How Has This Been Tested?

Created the device

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by klauer** [comment]
What a piece of work that pseudo-motor record is...

**Comment by ZLLentz** [comment]
In practice, this PR has a race condition where the normal metadata can bleed through and replace the corrected metadata. I'm going to solve this via creating an epics signal subclass that can be handed a metadata override dict that shadows the normal metadata dict, so regardless of the order of updates our corrected metadata is seen.

**Comment by ZLLentz** [comment]
Might need to rework the test suite of epics_motor a bit, it's a bit fragile to small changes like this.

File: pcdsdevices/signal.py [hunk 1]
```diff
@@ -640,3 +640,73 @@ def describe(self):
...
+                    f'Tried to override metadata key {key} in {self.name}, '
+                    'but this is not one of the metadata keys: '
+                    f'{self._metadata_keys}'
+                    )
+        try:
+            self._metadata_override.update(**md)
+        except AttributeError:
+            self._metadata_override = {}
+            self._metadata_override.update(**md)
+
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply pcdsdevices/signal.py:675]: I think I handled this correctly

**Comment by ZLLentz** [comment]
Nevermind, it turns out that when you override a component that uses, for example, a sub_value subscription, you need to re-establish the subscription in the subclass. See https://github.com/pcdshub/pcdsdevices/pull/783/commits/c7c2007ad62c79f99e6c5a2154413c2d345c146b.

**Comment by ZLLentz** [comment]
I think this is now complete

**Comment by ZLLentz** [comment]
If there are no objections, I'll merge before I stop for the day

**Review by klauer — APPROVED** [review]
Latest changes look good to me 👍

**Pull request merged**